src/s2/base/port.h

```diff
@@ -883,7 +883,7 @@ inline void UnalignedCopy64(const void *src, void *dst) {
 
 // __ASYLO__ platform uses newlib without an underlying OS, which provides
 // memalign, but not posix_memalign.
-#if defined(__cplusplus) &&                                             \
+#if !defined(__MINGW32__) && defined(__cplusplus) &&                    \
     (((defined(__GNUC__) || defined(__APPLE__) ||                  \
        defined(__NVCC__)) &&   \
       !defined(SWIG)) ||                                                \
```

Comment:
Move !defined(__MINGW32__) to after !defined(SWIG). This keeps the exceptions together.